2024-09-26-typescript-effect/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import { Effect, Array, Option, pipe, Stream, Chunk, StreamEmit } from "effect";
+import { Schema } from "@effect/schema";
+import {
+  FetchHttpClient,
+  HttpClient,
+  HttpClientResponse,
+} from "@effect/platform";
+
+const CityResponse = Schema.Struct({
+  name: Schema.String,
+  country_code: Schema.String.pipe(Schema.length(2)),
+  latitude: Schema.Number,
+  longitude: Schema.Number,
+});
+
+type CityResult = Schema.Schema.Type<typeof CityResponse>;
+
+const GeocodingResponse = Schema.Struct({
+  results: Schema.Array(CityResponse),
+});
+
+type GeocodingResponse = Schema.Schema.Type<typeof GeocodingResponse>;
+
+const Weather = Schema.Struct({
```

Comment:
should be `WeatherResponse` per blogpost

Reply:
Done!